did_core/did_methods/did_resolver_sov/src/resolution/utils.rs

```diff
@@ -69,50 +91,71 @@ pub(super) async fn ledger_response_to_ddo(
     log::info!("ledger_response_to_ddo >> did: {did}, verkey: {verkey}, resp: {resp}");
     let (service_id, ddo_id) = prepare_ids(did)?;
 
-    let service_data = get_data_from_response(resp)?;
-    log::info!("ledger_response_to_ddo >> service_data: {service_data:?}");
-    let endpoint: EndpointDidSov = serde_json::from_value(service_data["endpoint"].clone())?;
-
-    let txn_time = get_txn_time_from_response(resp)?;
-    let datetime = unix_to_datetime(txn_time);
-
-    let service_types: Vec<ServiceType> = endpoint
-        .types
-        .into_iter()
-        .map(|t| match t {
-            DidSovServiceType::Endpoint => ServiceType::AIP1,
-            DidSovServiceType::DidCommunication => ServiceType::DIDCommV1,
-            DidSovServiceType::DIDComm => ServiceType::DIDCommV2,
-            DidSovServiceType::Unknown => ServiceType::Other("Unknown".to_string()),
-        })
-        .collect();
-    let service = Service::new(
-        service_id,
-        endpoint.endpoint,
-        OneOrList::List(service_types),
-        Default::default(),
-    );
+    let service_data = match get_data_from_response(resp) {
+        Ok(data) => data,
+        Err(e) => {
+            log::warn!("Failed to get service data: {}", e);
+            serde_json::Value::Null
+        }
+    };
+
+    let mut ddo = DidDocument::new(ddo_id.clone());
+
+    let mut services = Vec::new();
+
+    if !service_data.is_null() {
+        let endpoint: EndpointDidSov = serde_json::from_value(service_data["endpoint"].clone())?;
```

Comment:
Note for anyone else reviewing; to my surprise, this should not be prone to panics:

https://docs.rs/serde_json/latest/serde_json/value/enum.Value.html#method.index

thanks serde!